src/modules/api-client/utils.ts

```diff
@@ -0,0 +1,257 @@
+import { ServerText, TranslatedString, translation } from "@atb/translations";
+import { NextApiRequest, NextApiResponse } from "next";
+import bunyan from "bunyan";
+import { v4 as uuidv4 } from "uuid";
+
+export const logger = bunyan.createLogger({
+  name: "planner-web",
+  streams: [{ stream: process.stdout, level: "info" }],
+  serializers: {
+    req: (req: Request) => {
+      return {
+        method: req.method,
+        url: req.url,
+        headers: {
+          ...req.headers,
+          cookie: null,
+        },
+      };
+    },
+    res: bunyan.stdSerializers.res,
+    upstream: (up: Response) => {
+      return {
+        statusCode: up.status,
+        url: up.url,
+        statusText: up.statusText,
+      };
+    },
+    err: bunyan.stdSerializers.err,
+  },
+});
+
+export function logError(e: any) {
+  if (process.env.NODE_ENV === "development") {
+    console.error(e);
+  } else {
+    logger.error(e);
+  }
+}
+
+export function logApplicationError(
+  e: ApplicationError,
+  req: Request | NextApiRequest,
+  res: Response | NextApiResponse<any>
+) {
+  if (process.env.NODE_ENV === "development") {
+    console.error(e);
+  } else {
+    logger.error(
+      {
+        upstream: e.upstreamResponse,
+        res,
+        req,
+        err: e,
+      },
+      e.message
+    );
+  }
+}
+
+const externalHttpUrls = {
+  entur: "https://api.entur.io",
```

Comment:
How might i.e. a staging environment tie into this? Just expand on this list as needed?

Reply:
Yes, thinking this will be used as an env var in that case. But usually, you won't have the need to go to staging for these endpoints.